### PR TITLE
Abstract rollup addresses into its own package

### PR DESCRIPTION
**Pull request by PJColombo**
### Checklist

- [x] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
It builds on top of #523.

Moves the rollup addresses into their own package (`@blobscan/rollup`) so it can be consumed by the `web`  and `api` package

#### Motivation and Context (Optional)
The rollup addresses are now required in the web app to implement the rollup filters

### CODE REVIEW

**Review by xFJA — CHANGES_REQUESTED** [review]
Same issue than target branch PR https://github.com/Blobscan/blobscan/pull/523#pullrequestreview-2266053770

File: packages/rollups/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "@blobscan/rollups",
+  "private": true,
+  "version": "0.2.1",
```
Comment (PabloCastellano) [inline packages/rollups/package.json:4]: shouldn't it be 0.0.1 or 0.1.0?

File: packages/rollups/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+export function getAddressByRollup(
+  rollup: Rollup,
+  chainId: number
+): string | null {
+  const rollupMapping = ROLLUP_TO_ADDRESS_MAPPINGS.get(chainId);
+
+  if (!rollupMapping) {
+    return null;
```
Comment (PabloCastellano) [inline packages/rollups/src/index.ts:97]: this case should be tested

File: packages/rollups/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+export function getRollupByAddress(
+  address: string,
+  chainId: number
+): Rollup | null {
+  const normalizedAddress = address.toLowerCase();
+  const addressToRollupMapping = ADDRESS_TO_ROLLUP_MAPPINGS.get(chainId);
+
+  if (!addressToRollupMapping) {
+    return null;
```
Comment (PabloCastellano) [inline packages/rollups/src/index.ts:84]: this case should be tested

File: packages/rollups/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      ),
+    ]
+  )
+);
+
+export function getChainRollups(chainId: number): Rollup[] {
+  const addressToRollupMapping = ADDRESS_TO_ROLLUP_MAPPINGS.get(chainId);
+
+  if (!addressToRollupMapping) {
+    return [];
```
Comment (PabloCastellano) [inline packages/rollups/src/index.ts:70]: this case should be tested

File: packages/rollups/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
+import { Rollup } from "@blobscan/db/prisma/enums";
+
+export const ADDRESS_TO_ROLLUP_MAPPINGS: Map<
+  number,
+  Map<string, Rollup>
+> = new Map([
+  // Mainnet
+  [
+    1,
+    new Map([
```
Comment (PabloCastellano) [inline packages/rollups/src/index.ts:10]: Why are you using Map() now?

**Review by PabloCastellano — CHANGES_REQUESTED** [review]
Left some questions. Also the packages list in the [documentation](https://docs.blobscan.com/docs/codebase-overview) should be updated

File: packages/rollups/src/index.ts (hunk 5 above)
Comment (PJColombo) [reply packages/rollups/src/index.ts:10]: It offers the `entries()` function, which allows you to retrieve both the key and value simultaneously. This makes building inverse mappings from rollup to address more straightforward.

File: packages/rollups/package.json (hunk 1 above)
Comment (PJColombo) [reply packages/rollups/package.json]: Yes

**Pull request merged**